src/client/utils/dom.ts

```diff
@@ -528,19 +528,27 @@ export function isBlob (instance): boolean {
 }
 
 export function isLocation (instance): boolean {
-    if (instance instanceof nativeMethods.locationClass ||
-        nativeMethods.objectToString.call(instance) === '[object Location]')
-        return true;
+    if (!instance)
+        return false;
 
-    try {
-        // eslint-disable-next-line no-restricted-properties
-        return instance && typeof instance === 'object' && instance.href !== void 0 && instance.assign !== void 0;
-    }
-    catch (e) {
-        // NOTE: Try to detect cross-domain window location.
-        // A cross-domain location has no the "assign" function in Safari.
-        return instance.replace && (isSafari || !!instance.assign);
+    if (isIE || isSafari) {
+        let instanceCtor = null;
```

Comment:
I propose to move this part to separate non-exported function `isLocationByPrototype (instance)`

Reply:
Ok. How about the `...Proto` function name based on this: `instance.__proto__;`?

Non-exportable `isLocationByProto`:
```js
function isLocationByProto (instance): boolean {
    ...
}

export function isLocation (instance): boolean {
    ....
    if (isIE || isSafari)
        return isLocationByProto(instance);
    ....
}
```

Reply:
Ok. Let it be 'Proto'